content-service/src/main/java/io/meeds/news/rest/NewsRest.java

```diff
@@ -821,4 +827,57 @@ public Response deleteArticleTranslation(@PathVariable("id")
       return Response.serverError().entity(e.getMessage()).build();
     }
   }
+
+  @GetMapping(path = "translation/{articleId}", produces = MediaType.APPLICATION_JSON_VALUE)
+  @Secured("users")
+  @Operation(summary = "Get article available translation languages", method = "GET", description = "Get article available translation languages")
+  @ApiResponses(value = { @ApiResponse(responseCode = "200", description = "article version deleted"),
+          @ApiResponse(responseCode = "400", description = "Invalid query input"),
+          @ApiResponse(responseCode = "401", description = "User not authorized to delete the article"),
+          @ApiResponse(responseCode = "500", description = "Internal server error") })
+  public ResponseEntity<List<String>> getAvailableTranslationLanguages(@PathVariable("articleId")
+                                                                       String articleId,
+                                                                       @RequestParam(name = "withDrafts")
+                                                                       boolean withDrafts) {
+    try {
+      if (StringUtils.isBlank(articleId)) {
+        return ResponseEntity.badRequest().build();
+      }
+      return ResponseEntity.ok(newsService.getArticleLanguages(articleId, withDrafts));
+    } catch (Exception e) {
+      LOG.error("Error when getting the article available translation languages with id " + articleId, e);
+      return ResponseEntity.internalServerError().build();
+    }
+  }
+
+  @GetMapping(value = "/languages", produces = "application/json")
+  public ResponseEntity<String> getAvailableLanguages(@RequestParam(value = "lang", required = false) String lang) {
+    try {
+      Set<Locale> locales = LocaleContextInfoUtils.getSupportedLocales();
+      List<Locale> localesList = new ArrayList<>(locales);
+      JSONArray localesJSON = new JSONArray();
+      Locale currentLocal = Locale.ENGLISH;
+      if (!lang.isEmpty()) {
+        Optional<Locale> opLocal = Arrays.stream(Locale.getAvailableLocales())
+                .filter(local -> local.getLanguage().equals(lang))
+                .findAny();
+        if (opLocal.isPresent()) {
+          currentLocal = opLocal.get();
+        }
+      }
+      for (Locale locale : localesList) {
+        JSONObject object = new JSONObject();
+        if (locale == null || locale.toString().equals("ma")) {
+          continue;
+        } else {
+          object.put("value", locale.toString());
+          object.put("text", locale.getDisplayName(currentLocal));
+        }
+        localesJSON.add(object);
+      }
+      return ResponseEntity.ok(localesJSON.toString());
```

Comment:
I am not sure if this will be the best option, The list of languages available in the platform should be available for all users statically as the list does not get changed.
Could you add it like in [this JSP](https://github.com/Meeds-io/social/blob/develop/webapp/portlet/src/main/webapp/WEB-INF/jsp/portlet/userSettingLanguage.jsp#L11) instead of adding a Rest endpoint for it ?

Reply:
available languages  defined as eXo variable!